world/LevelFuncMgr.cpp

```diff
@@ -72,16 +74,15 @@ void LevelFuncMgr::DestroyLevelFunc(LevelFunc *pFunc)
 
     RemoveFromActiveList(pFunc);
 
-    std::vector<LevelFunc *>::iterator iter = m_FuncList.begin();
-    std::vector<LevelFunc *>::iterator end  = m_FuncList.end();
+    const auto iter = std::find_if(m_FuncList.begin(), m_FuncList.end(),
+        [pFunc](const LevelFunc *levelFunc)
+    {
+        return levelFunc == pFunc;
+    });
```

Comment:
FWIW, something simple like this, comparing each container element to a value, works just fine with `std::find`.

Reply:
Oh, whoops, I forgot. I was in `std::find_if()` mode. They have the same result, though.